gensim/test/test_doc2vec.py

```diff
@@ -95,6 +100,13 @@ def testPersistenceWord2VecFormat(self):
         binary_model_dv = keyedvectors.KeyedVectors.load_word2vec_format(test_word, binary=True)
         self.assertEqual(len(model.wv.vocab), len(binary_model_dv.vocab))
 
+    def test_unicode_in_doctag(self):
```

Comment:
Please add a test with an exception (and check it with assertraise).

Reply:
The test should rather be the inverse of assertraise. The exception should not be thrown with this PR. I will add a inverse test.

Reply:
It's a good idea to add 2 tests: first with `unicode_tags=False` and assertRaise and second is a current test.

Reply:
I think you are mixing up what really happens, let me explain:
In gensim 2.1 a modified `save_word2vec_format` for Doc2Vec was added in #1256. The storing handled to export any utf8 symbols [doc2vec.py#L853](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/doc2vec.py#L853). Unfortunately, the `str(..)` call in [doc2vec.py#L850](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/doc2vec.py#L850) fails in Python 2.7 when the doctags contain chars not ASCII encodable.
Currently the develop branch is broken and the added test will fail, i.e. `UnicodeEncodeError` is thrown for Python 2.7.
The test now checks whether it can successfully store unicode doctags and fails otherwise. I have to generate them (with the code above) since the default string doctags are ASCII encodable (-> addition of the '¡' character).
So, what should the second test actually be then?